apollo-router/src/services/router_service.rs

```diff
@@ -402,7 +402,7 @@ impl RouterService {
         if results.len() == 1 {
             Ok(results.pop().expect("we should have at least one response"))
         } else {
-            let first = results.pop().expect("we should have at least one response");
+            let first = results.remove(0);
```

Comment:
`remove()` will shift all the array elements. Inset you can do something like:

```rust
let mut it = results.into_iter();
let first = it.next();

for result in it {
```

Reply:
done: 2958874eda037951a5798e7c7f4fd0e9e743039f